src/physics/lsm_driver.f90

```diff
@@ -874,9 +1034,72 @@ subroutine lsm(domain,options,dt)
+                    ,lakedepth2d=lakedepth2d                                    &
+                    ,watsat3d=watsat3d      ,csol3d=csol3d      ,tkmg3d=tkmg3d  &
+                    ,tkdry3d=tkdry3d        ,tksatu3d=tksatu3d                  &
+                    ,ivgtyp=domain%veg_type                                     &
+                    ,HT=domain%terrain%data_2d                                  &
+                    ,xland=real(domain%land_mask)                               & !-- XLAND         land mask (1 for land, 2 OR 0 for water)  i/o
+                    ,iswater=iswater,  xice=xice,   xice_threshold=xice_threshold   &
+                    ,lake_min_elev=5.                                           & ! if this value is changed, also change it in lake_ini
+                    ,ids=ids, ide=ide, jds=jds, jde=jde, kds=kds, kde=kde       &
+                    ,ims=ims, ime=ime, jms=jms, jme=jme, kms=kms, kme=kme       &
+                    ,its=its, ite=ite, jts=jts, jte=jte, kts=kts, kte=kte       &
+                    ,h2osno2d=h2osno2d,         snowdp2d=snowdp2d               &
+                    ,snl2d=snl2d               ,z3d=z3d                         &
+                    ,dz3d=dz3d                 ,zi3d=zi3d                       &
+                    ,h2osoi_vol3d=h2osoi_vol3d ,h2osoi_liq3d=h2osoi_liq3d       &
+                    ,h2osoi_ice3d=h2osoi_ice3d ,t_grnd2d=t_grnd2d               &
+                    ,t_soisno3d=t_soisno3d   ,t_lake3d=t_lake3d                 &
+                    ,savedtke12d=savedtke12d  ,lake_icefrac3d=lake_icefrac3d    &
+                    ,lakemask=lakemask                                          &
+                    ,lakeflag=lakeflag                                          &
+                    ,hfx= domain%sensible_heat%data_2d                          & !(OUT)-- HFX         upward heat flux at the surface (W/m^2)   (INTENT:OUT)
+                    ,lh=domain%latent_heat%data_2d                              & !(OUT)-- LH          net upward latent heat flux at surface (W/m^2)
+                    ,grdflx=domain%ground_heat_flux%data_2d                     & !(OUT)-- GRDFLX(I,J) ground heat flux (W m-2)
+                    ,tsk=domain%skin_temperature%data_2d                        & !(OUT)-- TSK          skin temperature [K]
+                    ,qfx=QFX                                                    & !(OUT)-- QFX        upward moisture flux at the surface (kg/m^2/s) in 
+                    ,t2= domain%temperature_2m%data_2d                          & !(OUT)-- t2         diagnostic 2-m temperature from surface layer and lsm
+                    ,th2=TH2                                                    & !(OUT)-- th2        diagnostic 2-m theta from surface layer and lsm
+                    ,q2=domain%humidity_2m%data_2d                              & !(OUT)-- q2         diagnostic 2-m mixing ratio from surface layer and lsm
```

Comment:
are these really all just intent(out)? it says T2 TH2, Q2 are from the LSM+surface layer... so those should be intent(in) or maybe (inout)? I wonder if we need to compute TH2 to pass it in then? (could be computed from temperature_2d)

Reply:
Any variable_t variables that are being used here, should also be requested for allocation in the alloc_vars routine for the lake model

Reply:
Subroutine Lake lists them as OUT. Why would they be IN? 

(added the vars to the alloc_vars, sorry!)